src/java/org/apache/cassandra/db/ReadCommand.java

```diff
@@ -466,6 +466,8 @@ public UnfilteredPartitionIterator executeLocally(ReadExecutionController execut
                 iterator = withQuerySizeTracking(iterator);
                 iterator = maybeSlowDownForTesting(iterator);
                 iterator = withQueryCancellation(iterator);
+                if (DatabaseDescriptor.getPurgeableTobmstonesMetricGranularity() != Config.TombstonesMetricGranularity.disabled)
```

Comment:
for readability's sake, I would do `maybeRecordPurgeableTombstones` and in that method there would be this `if` and if not true then I would just return `iter`. The same way it is done for `maybeSlowDownForTest` two lines above.